integration-tests/tests/src/tests/queries.ts

```diff
@@ -150,6 +197,92 @@ const expectQueryResultValues = (
 };
 
 describe("Queries", () => {
+  describe("GeoSpatial", () => {
+    openRealmBeforeEach({ schema: [PointOfInterest, MyGeoPoint.schema] });
+
+    const zero: IPointOfInterest = {
+      id: 1,
+      location: new MyGeoPoint(0, 0),
+    };
+
+    const poiA: IPointOfInterest = {
+      id: 2,
+      location: new MyGeoPoint(50.5, 50.5),
+    };
+
+    const poiB: IPointOfInterest = {
+      id: 3,
+      location: new MyGeoPoint(40, -40),
+    };
+
+    const poiC: IPointOfInterest = {
+      id: 4,
+      location: new MyGeoPoint(-30, -30.5),
+    };
+
+    const poiD: IPointOfInterest = {
+      id: 5,
+      location: new MyGeoPoint(-25, 25),
+    };
+
+    function convertIGeoPosition(point: IGeoPosition): string {
+      return `[${point[0]}, ${point[1]}]`;
+    }
+
+    function convertGeoPoint(point: GeoPoint): string {
+      if (Array.isArray(point)) {
+        return convertIGeoPosition(point);
+      }
+
+      if ("latitude" in point) {
+        return convertIGeoPosition([point.longitude, point.latitude]);
+      }
+
+      return convertIGeoPosition(point.coordinates);
+    }
+
+    function convertGeoCircle(circle: GeoCircle): string {
+      return `geoSphere(${convertGeoPoint(circle.center)},  ${circle.distance})`;
+    }
+
+    function circleTest(realm: Realm, circle: GeoCircle, pois: PointOfInterest[]) {
+      const circleString = convertGeoCircle(circle);
+      geoTest(realm, circle, circleString, pois);
+    }
+
+    function geoTest(realm: Realm, geo: GeoCircle | GeoBox | GeoPolygon, geoString: string, pois: PointOfInterest[]) {
+      const poiIds = pois.map((p) => p.id);
+      expectQueryResultValues(realm, PointOfInterest, "id", [[poiIds, "location geoWithin $0 SORT(id ASC)", geo]]);
+      expectQueryResultValues(realm, PointOfInterest, "id", [[poiIds, `location geoWithin ${geoString} SORT(id ASC)`]]);
+    }
+
+    beforeEach(function (this: RealmContext) {
+      this.realm.write(() => {
+        this.realm.create("PointOfInterest", zero);
+        this.realm.create("PointOfInterest", poiA);
+        this.realm.create("PointOfInterest", poiB);
+        this.realm.create("PointOfInterest", poiC);
+        this.realm.create("PointOfInterest", poiD);
+      });
+    });
+
+    it.only("GeoCircle basic", function (this: RealmContext) {
```

Comment:
```suggestion
    it("GeoCircle basic", function (this: RealmContext) {
```

Reply:
Oh yes, this is only for now 😁 
Just a fast draft